### PR TITLE
Add tests for export job admin progress

### PR DESCRIPTION
**Pull request by Eg0ra**

### CODE REVIEW

File: import_export_extensions/models/export_job.py [hunk 1]
```diff
@@ -159,13 +159,6 @@ def export_filename(self) -> str:
     def progress(self) -> TaskStateInfo | None:
         """Return dict with parsing state.
 
-        Example for sync mode::
```
Comment (NikAzanov) [inline import_export_extensions/models/export_job.py:162]: Let's also update part about "celery (celery) mode". I guess now it's relevant for any mode, so just remove `celery` mentions.

Also I see "Possible states" in docstring, but can we also add this enum to `TaskStateInfo`?

File: test_project/tests/integration_tests/test_admin/test_export.py [hunk 2]
```diff
@@ -49,3 +52,123 @@ def test_export_using_admin_model(client: Client, superuser: User):
     assert ExportJob.objects.exists()
     export_job = ExportJob.objects.first()
     assert export_job.export_status == ExportJob.ExportStatus.EXPORTED
+
+
+@pytest.mark.django_db(transaction=True)
+def test_export_progress_for_sync_mode(
```
Comment (NikAzanov) [inline test_project/tests/integration_tests/test_admin/test_export.py:58]: Is there a difference between sync/async modes now?

Please, provide more detailed docstring (why there are two different tests for sync and async and how it works)

File: test_project/tests/integration_tests/test_admin/test_export.py [hunk 3]
```diff
@@ -49,3 +52,123 @@ def test_export_using_admin_model(client: Client, superuser: User):
...
+):
+    """Test export job admin progress page."""
+    client.force_login(superuser)
+
+    # Prepare data to imitate intermediate task state
+    mocker.patch(
+        "celery.result.AsyncResult.info",
+        new={"current": 2, "total": 3},
+    )
+    expected_percent = 66
```
Comment (NikAzanov) [inline test_project/tests/integration_tests/test_admin/test_export.py:92]:
```suggestion
    fake_progress_info = {
        "current": 2,
        "total": 3,
    }
    mocker.patch(
        "celery.result.AsyncResult.info",
        new=fake_progress_info,
    )
    expected_percent = int(fake_progress_info["current"] / fake_progress_info["total"])
```

File: test_project/tests/integration_tests/test_admin/test_export.py [hunk 4]
```diff
@@ -49,3 +52,123 @@ def test_export_using_admin_model(client: Client, superuser: User):
...
+        path=reverse(
+            "admin:export_job_progress",
+            kwargs={"job_id": artist_export_job.pk},
+        ),
+    )
+    assert response.status_code == status.HTTP_200_OK
+    json_data = response.json()
+    assert "total" in json_data
+    assert "current" in json_data
+    assert response.json()["percent"] == expected_percent
```
Comment (NikAzanov) [inline test_project/tests/integration_tests/test_admin/test_export.py:107]: Why `response.json()` again?

Maybe prepare an object like `{"percent": expected_percent, **fake_progress_info}` and compare with it?

File: test_project/tests/integration_tests/test_admin/test_export.py [hunk 5]
```diff
@@ -49,3 +52,123 @@ def test_export_using_admin_model(client: Client, superuser: User):
...
+        path=reverse(
+            "admin:export_job_progress",
+            kwargs={
+                "job_id": job_id,
+            },
+        ),
+    )
+
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json()["validation_error"] == expected_error_message
```
Comment (NikAzanov) [inline test_project/tests/integration_tests/test_admin/test_export.py:136]: image

File: test_project/tests/integration_tests/test_admin/test_export.py [hunk 6]
```diff
@@ -49,3 +52,123 @@ def test_export_using_admin_model(client: Client, superuser: User):
...
+    """Test export job admin progress page with deleted export job."""
+    client.force_login(superuser)
+
+    mocker.patch(
+        "celery.result.AsyncResult.state",
+        new=states.FAILURE,
+    )
+    mocker.patch(
+        "celery.result.AsyncResult.info",
+        new="Mocked Error Message",
```
Comment (NikAzanov) [inline test_project/tests/integration_tests/test_admin/test_export.py:154]: Can `AsyncResult.info` be a string? 

I guess we'll need to improve typing (`TaskStateInfo`)

File: test_project/tests/integration_tests/test_admin/test_export.py [hunk 7]
```diff
@@ -49,3 +52,123 @@ def test_export_using_admin_model(client: Client, superuser: User):
...
+    assert response.json()["validation_error"] == expected_error_message
+
+
+@pytest.mark.django_db(transaction=True)
+def test_export_progress_with_failed_celery_task(
+    client: Client,
+    superuser: User,
+    mocker: pytest_mock.MockerFixture,
+):
+    """Test export job admin progress page with deleted export job."""
```
Comment (NikAzanov) [inline test_project/tests/integration_tests/test_admin/test_export.py:145]: Please improve docstring with the purpose of test (check than after celery fail there will be export error)

File: test_project/tests/test_import_job/test_import_data.py [hunk 8]
```diff
@@ -156,3 +158,26 @@ def test_import_data_with_validation_error(existing_artist: Artist):
...
+    """Test import job max dataset rows validation."""
+    import_job: ImportJob = ArtistImportJobFactory(
+        artists=[new_artist],
+        skip_parse_step=True,
+        is_valid_file=False,
+    )
+    expected_error_message = (
+        "Too many rows `2`(Max: 1). Input file may be broken. If it's "
+        "spreadsheet file, please delete empty rows."
+    )
```
Comment (NikAzanov) [inline test_project/tests/test_import_job/test_import_data.py:178]: I'm not sure that hardcoding value is a good practice. Maybe just check `f"Too many rows `2`" in import_job.error_message`? Or the best way is to move error message in a constance and then just reuse in tests

**Review by NikAzanov — CHANGES_REQUESTED** [review]

**Review by OttoAndrey — APPROVED** [review]

**Review by NikAzanov — APPROVED** [review]

**Review by NikAzanov — APPROVED** [review]

File: import_export_extensions/admin/model_admins/export_job_admin.py [hunk 9]
```diff
@@ -1,4 +1,6 @@
 
+from http import HTTPStatus
```
Comment (TheSuperiorStanislav) [inline import_export_extensions/admin/model_admins/export_job_admin.py:2]:
```suggestion
import http

http.HTTPStatus.NOT_FOUND
```
Just personal preference

**Review by TheSuperiorStanislav — APPROVED** [review]

**Pull request merged**